pkg/cluster/cluster.go

```diff
@@ -74,6 +77,10 @@ type manager struct {
 	samplescli    samplesclient.Interface
 	securitycli   securityclient.Interface
 	arocli        aroclient.AroV1alpha1Interface
+
+	newDynamicValidator   func(*logrus.Entry, env.Interface, *api.OpenShiftCluster, *api.SubscriptionDocument, refreshable.Authorizer) validate.OpenShiftClusterDynamicValidator
+	getClusterSPObjectID  func(context.Context, env.Interface, *logrus.Entry, *api.OpenShiftClusterDocument) (string, error)
+	getBlobServiceBackend func(context.Context, storage.AccountsClient, *api.OpenShiftCluster, mgmtstorage.Permissions, mgmtstorage.SignedResourceTypes) (blob.BlobService, error)
```

Comment:
What's the motivation behind these changes? Is the plan also to mock these fields for unit testing 
 purposes, like `newDynamicValidator`? AFAICT, they are only instantiated in `pkg/cluster/cluster.go`, and not mocked in any existing tests:

```
$ grep -rniI "getblobservicebackend:\|getclusterspobjectid:" * 
pkg/cluster/cluster.go:134:             getClusterSPObjectID:  clusterSPObjectID,
pkg/cluster/cluster.go:135:             getBlobServiceBackend: blob.NewBlobServiceForCluster,
```

Just wondering if these changes are necessary.